File: elabdufrontend/src/services/OrderService.js
import axios from 'axios'

const USER_BASE_REST_API_URL = 'http://localhost:8083/orders';
class OrderService {
    saveOrder(order) {
        return axios.post(USER_BASE_REST_API_URL , order)
    }

    getOrdersForCustomer(customerId) {
        return axios.get(USER_BASE_REST_API_URL + "/" + customerId)
    }

    getOrderDetailsForOrder(orderId) {
        return axios.get(USER_BASE_REST_API_URL + "/details" + orderId)
    }

    getAllOrder() {
        return axios.get(USER_BASE_REST_API_URL );
    }

    deleteOrder(orderId) {
        return axios.delete(USER_BASE_REST_API_URL + "/" + orderId);
    }
    // getAllUser() {
    //     return axios.get(USER_BASE_REST_API_URL + "/");
    // }

    // getUserById(id) {
    //     return axios.get(USER_BASE_REST_API_URL + "/" + id);
    // }

    // getUserByEmail(email) {
    //     return axios.get(USER_BASE_REST_API_URL + "/getUser" + email);
    // }

    // deleteUser(id) {
    //     return axios.delete(USER_BASE_REST_API_URL + "/delete/" + id);
    // }

    // updateUser(id, user) {
    //     return axios.put(USER_BASE_REST_API_URL + "/update/" + id, user);
    // }
}
export default new OrderService();

File: ordersBackend/src/main/java/com/example/usermetadata/service/OrderService.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.example.usermetadata.service;

import com.example.usermetadata.data.Order;
import com.example.usermetadata.data.OrderRepo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

/**
 * @author User
 */
@Service
public class OrderService implements OrderServiceCo {
    @Autowired
    private OrderRepo orderRepo;

    @Override
    public Order createOrder(Order order) {
        return orderRepo.save(order);
    }

    @Override
    public List<Order> getAllUOrders() {
        return orderRepo.findAll();
    }

    @Override
    public List<Order> getAllUOrdersByCustomerId(int customerId) {
        return orderRepo.findByCustomerId(customerId);
    }

//    @Override
//    public List<UserMetaData> getAllUser() {
//        return orderRepo.findAll();
//    }
//
//    @Override
//    public UserMetaData getUserById(int id) {
//        return (UserMetaData) orderRepo.findById(id);
//    }
//
//    @Override
//    public List<UserMetaData> getUserByEmail(String email) {
//        return  orderRepo.findByEmail(email);
//    }
//
//    @Override
//    public void deleteUser(int id) {
//        UserMetaData user = (UserMetaData) orderRepo.findById(id);
//        if (user != null) {
//            orderRepo.delete(user);
//        }
//    }
//
//
//    @Override
//    public UserMetaData updateUser(int id, UserMetaData user) {
//        return orderRepo.save(user);
//    }

}
